rskj-core/src/test/java/co/rsk/peg/BridgeSupportTest.java

```diff
@@ -5847,34 +5848,151 @@ void validationsForRegisterBtcTransaction_successful() throws IOException, Block
         Assertions.assertTrue(bridgeSupport.validationsForRegisterBtcTransaction(tx.getHash(), height, pmt.bitcoinSerialize(), tx.bitcoinSerialize()));
     }
 
+    private static Stream<Arguments> getTransactionType_sentFromP2SHErpFed_Args() {
+        return Stream.of(
+            // It should be considered a PEGOUT
+            Arguments.of(ActivationConfigsForTest.hop400().forBlock(0), PegTxType.PEGOUT_OR_MIGRATION),
+            // It should be considered a PEGOUT
+            Arguments.of(ActivationConfigsForTest.hop401().forBlock(0), PegTxType.PEGOUT_OR_MIGRATION)
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("getTransactionType_sentFromP2SHErpFed_Args")
+    void getTransactionType_sentFromP2SHErpFed(ActivationConfig.ForBlock activations, PegTxType expectedTxType) {
+        // Arrange
+        BridgeStorageProvider provider = mock(BridgeStorageProvider.class);
+        BridgeSupport bridgeSupport = bridgeSupportBuilder
+            .withBridgeConstants(bridgeConstantsRegtest)
+            .withActivations(activations)
+            .withProvider(provider)
+            .build();
+
+        Federation activeFederation = bridgeConstantsRegtest.getGenesisFederation();
+
+        when(provider.getNewFederation()).thenReturn(activeFederation);
+
+        List<BtcECKey> emergencyKeys = PegTestUtils.createRandomBtcECKeys(3);
+        long activationDelay = 256L;
```

Comment:
I'd get these values from BridgeConstants, to make it more realistic. I would also consider using BridgeMainnetConstants by default, to make it even more realistic